client/components/plans/plan-price/index.jsx

```diff
@@ -23,8 +23,10 @@ const PlanPrice = React.createClass( {
 				return this.translate( 'Free', { context: 'Zero cost product price' } );
 			}
 
-			const currencySymbol = formattedPrice.slice( 0, 1 );
-			const monthlyPrice = +( rawPrice / 12 ).toFixed( currencySymbol === '¥' ? 0 : 2 );
+			const dollarIndex = formattedPrice.indexOf( '$' );
+			const symbolIndex = dollarIndex > -1 ? dollarIndex + 1 : 1;
+			const currencySymbol = formattedPrice.slice( 0, symbolIndex ); // A$ or C$
+			const monthlyPrice = ( rawPrice / 12 ).toFixed( currencySymbol === '¥' ? 0 : 2 );
```

Comment:
This is starting to look a bit complicated. Could we break it into more semantic pieces?

```js
// could get $5.95, A$4.13, ¥298, €3,50, etc…
const splitFormattedPrice = p => compact( split( p, /([^\d]+)(.*)/ ) );
const currencyDigits = prefix => get( [
	'¥': 0
], prefix, 2 );

const [ prefix, rawPrice ] = splitFormattedPrice( formattedPrice );
const annualPrice = Number.parseFloat( rawPrice );
const monthlyPrice = ( annualPrice / 12 ).toFixed( currencyDigits( prefix ) );

return `${ prefix }${ monthlyPrice }`
```

[jsbin test](http://jsbin.com/geqicojomu/5/edit?js,console)

With this we're a bit more flexible with what formats we get (such as use of commas instead of periods for separators) and it's extensible with the digits. Hopefully that flow is clearer as well what computation is actually going on.